test/fs/test_fs_js_api.c

```diff
@@ -37,6 +37,39 @@ int main() {
 #endif
     );
 
+    /********** test FS.rename() **********/
+    EM_ASM(
+        FS.mkdir('renamedir');
+        FS.writeFile('renamedir/renametestfile', "");
+        FS.writeFile('toplevelfile', "");
+        FS.rename('renamedir/renametestfile', 'renamedir/newname');
+        var newnameStream = FS.open('renamedir/newname', 'r');
+        assert(newnameStream);
+
+        var ex;
+        try {
+            FS.open('renamedir/renametestfile', 'r');
+        } catch (err) {
+            ex = err;
+        }
+        assert(ex.name === "ErrnoError" && ex.errno === 44 /* ENOENT */);
+
+        
+        try {
+            FS.rename('renamedir', 'renamedir/newname');
+        } catch (err) {
+            ex = err;
+        }
+        assert(ex.name === "ErrnoError" && ex.errno === 28 /* EINVAL */);
```

Comment:
Maybe add a comment about why EINVAL here (IIUC, because we rename a file to a child of itself, which is not valid?)